crates/subspace-archiving/src/archiver.rs

```diff
@@ -725,6 +719,34 @@ impl Archiver {
             pieces
         };
 
+        let existing_commitments = self.incremental_record_commitments.len();
+        // Add commitments for pieces that were not created incrementally
+        {
+            #[cfg(not(feature = "rayon"))]
+            let source_pieces = pieces.source();
+            #[cfg(feature = "rayon")]
+            let source_pieces = pieces.par_source();
+
+            let iter = source_pieces
+                .skip(existing_commitments)
+                .map(|piece| piece.record().safe_scalar_arrays().map(Scalar::from))
+                .map(|scalars| {
+                    let polynomial = self.kzg.poly(&scalars.collect::<Vec<_>>()).expect(
+                        "KZG instance must be configured to support this many scalars; qed",
+                    );
```

Comment:
The CI did not contain the necessary message for where `extract_core_domain_wasm_bundle_in_system_domain_runtime_should_work` failed on, but it can be reproduced in my machine, which shows this expect is panicked:
```
thread '<unnamed>' panicked at 'KZG instance must be configured to support this many scalars; qed: "A list with power-of-two length expected"'
```
the test probably failed because it ran a primary node.